storage/series.go

```diff
@@ -128,11 +138,83 @@ func (c *chunkSetToSeriesSet) Err() error {
 	return c.ChunkSeriesSet.Err()
 }
 
-type chunkToSeries struct {
+type chunkToSeriesDecoder struct {
+	chunks.Meta
+
 	labels labels.Labels
-	chk    chunks.Meta
-	buf    chunkenc.Iterator
 }
 
-func (s *chunkToSeries) Labels() labels.Labels       { return s.labels }
-func (s *chunkToSeries) Iterator() chunkenc.Iterator { return s.chk.Chunk.Iterator(s.buf) }
+func (s *chunkToSeriesDecoder) Labels() labels.Labels { return s.labels }
+
+// TODO(bwplotka): Can we provide any chunkenc buffer?
+func (s *chunkToSeriesDecoder) Iterator() chunkenc.Iterator { return s.Chunk.Iterator(nil) }
+
+type seriesSetToChunkSet struct {
+	SeriesSet
+}
+
+// NewSeriesSetToChunkSet converts SeriesSet to ChunkSeriesSet by encoding chunks from samples.
+func NewSeriesSetToChunkSet(chk SeriesSet) ChunkSeriesSet {
+	return &seriesSetToChunkSet{SeriesSet: chk}
+}
+
+func (c *seriesSetToChunkSet) Next() bool {
+	if c.Err() != nil || !c.SeriesSet.Next() {
+		return false
+	}
+	return true
+}
+
+func (c *seriesSetToChunkSet) At() ChunkSeries {
+	return &seriesToChunkEncoder{
+		Series: c.SeriesSet.At(),
+	}
+}
+
+func (c *seriesSetToChunkSet) Err() error {
+	return c.SeriesSet.Err()
+}
+
+type seriesToChunkEncoder struct {
+	Series
+}
+
+// TODO(bwplotka): Currently encoder will just naively build one chunk, without limit. Split it: https://github.com/prometheus/tsdb/issues/670
+// DO IT NOW.
```

Comment:
:thinking: ?

Reply:
XD talking to myself :man_facepalming: 

will remove.